src/parser/expr.js

```diff
@@ -508,6 +508,23 @@ module.exports = {
     return expr;
   },
 
+  /**
+   * Recursively convert nested array to nested list.
+   */
+  convertToList: function (items) {
+    const convertedItems = items.map((entry) => {
+      if (
+        entry.value &&
+        entry.value.kind === "array" &&
+        entry.value.shortForm
+      ) {
+        entry.value = this.convertToList(entry.value.items);
+      }
+      return entry;
+    });
+    return this.node("list")(convertedItems, true);
```

Comment:
@czosel this.node("...") => creates a node from the parsing context, should be used only when consuming tokens - as items were already processed, you need on created node to get in sync with locations & comments (available on array node)

```
convertToList: function (array) {
    const convertedItems = array.items.map((entry) => {
      if (
        entry.value &&
        entry.value.kind === "array" &&
        entry.value.shortForm
      ) {
        entry.value = this.convertToList(entry.value);
      }
      return entry;
    });
    const node = this.node("list")(convertedItems, true);
    node.loc = array.loc;
    node.leadingComments = array.leadingComments;
    node.trailingComments = array.trailingComments;
    return node;
```

Reply:
Hi @ichiriac, thanks for the detailed feedback! I made the change that fixes the `loc` issue, but I left out the comment part because I wasn't able to write a test that would show an issue with comments. Are you sure it is really needed? If yes, can you post an example code snipped that should show the issue? I tried `/* test */ [$a, [$b, $c]] = [1,[2,3]]` and `[$a, /* test */ [$b, $c]] = [1,[2,3]]`. Thanks again!

Reply:
Hi @czosel, did you tries to enable comments flags by default the parser does not extracts comments. In principle without these statements comments eaten from node will not be available on created one.

Also try this one : `[$a, /* test */ [$b, $c]] =  [1,[2,3]] /* should not be eaten by newly created node ! */`

Reply:
hi @czosel, still missing the comments statements, waiting for these before merging, other parts are clean

Reply:
Hi @ichiriac, sorry about the delay - I hope I’ll find the time to finalize it this weekend!

Reply:
no problemo dude, take your time, and tell me when you're ready to merge

Reply:
Hi @ichiriac, I managed to take another look. I wasn't able to write a test for this because comments in our example either get assigned to `entry` or `assign` nodes, but never to `list`. I added the comment assignment as you suggested anyway - maybe there are situations where comments get attached to the list; and even if there currently is no such case, it possibly prevents future bugs if the comment handling ever changes :slightly_smiling_face: 

Let me know if you'd still change anything about this!

Reply:
thanks @czosel - I don't see also any usecase for now but your code is more solid this way as it covers better the api 😄